src/feature-manager.ts

```diff
@@ -176,33 +176,48 @@ const add = async (
       setupPageLoad(id, details);
     }
 
+    /**
+     * debounce function helps to control the frequency at which a function is triggered in
+     * response to an event. It ensures that the action is executed only once after a certain
+     * duration of inactivity, even if the event is fired multiple times within that duration.
+     *  */
+    const debounce = function (func: Function, delay: number) {
+      let debounceTimer: any;
+      return function () {
+        clearTimeout(debounceTimer);
+        debounceTimer = setTimeout(func, delay);
+      };
+    };
```

Comment:
Hi @wxharry, the [debounce helper](https://github.com/hypertrons/hypertrons-crx/blob/94bcf7231b7c82db6f477f7859c87f5d8b5c3a5f/src/helpers/debounce.ts) is already in the code base :D

Reply:
Oh, I didn't notice that. Will replace with the existing code